src/interface/swarm.cpp

```diff
@@ -47,6 +71,49 @@ Swarm::Swarm(const std::string &label, const Metadata &metadata, const int nmax_
   marked_for_removal_.data.DeepCopy(marked_for_removal_h);
 }
 
+bool Swarm::FinishCommunication(BoundaryCommSubset phase) {
+  printf("[%i] FinishCommunication\n", Globals::my_rank);
+  GetBlockPointer()->exec_space.fence();
+
+  if (allreduce_request_ == MPI_REQUEST_NULL) { // No outstanding Iallreduce request
+    printf("NULL!\n");
+    MPI_Iallreduce(&local_num_completed_, &global_num_completed_, 1, MPI_INT, MPI_SUM,
+                   MPI_COMM_WORLD, &allreduce_request_);
+  } else { // Outstanding Iallreduce request
+    int flag;
+    MPI_Test(&allreduce_request_, &flag, MPI_STATUS_IGNORE);
+    printf("Not NULL! flag: %i\n", flag);
+    if (flag) { // Iallreduce completed
+
+      // TODO(BRR) temporary!
+      mpiStatus = true;
+      return true;
+
+      printf("[%i] incomplete: %i completed: %i\n", Globals::my_rank,
+             global_num_incomplete_, global_num_completed_);
+
+      /*if (global_num_incomplete_ > global_num_completed) { // Transport not done
+        return false;
+      } else { // Transport completed
+        mpiStatus = true;
+        return true;
+      }*/
+
+      // TODO(BRR) change the name of these vars
+      printf("incomp: %i comp: %i\n", global_num_incomplete_, global_num_completed_);
+      if (global_num_incomplete_ == global_num_completed_) {
+        // Transport completed
+        mpiStatus = true;
+        return true;
+      }
+
+      allreduce_request_ = MPI_REQUEST_NULL;
+    }
+  }
+
+  return false;
+}
```

Comment:
This is obfuscated because `Swarm::StartCommunication` is defined in the header file, but I think that's what I do at least in terms of using two tasks. Agreed that I need to switch to meshdata